### PR TITLE
Use planemo-discover-action to reduce a bit of CI boilerplate

### PR DESCRIPTION
**Pull request by mvdbeek**
FOR CONTRIBUTOR:
* [ ] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by mvdbeek** [comment]
@bernt-matthias @nsoranzo @bgruening what do you think about this ?
Ideally we'd also push the caching and artifact up/download into the composite action, but they don't support steps with `using:` yet. If we like this I would rename `mvdbeek/planemo-discover-action` into `galaxyproject/planemo-action` or something along the lines

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -109,31 +79,22 @@ jobs:
...
-      run: pip install planemo
-    - name: Planemo lint
-      run: |
-        set -e
-        while read -r DIR; do
-            planemo shed_lint --tools --ensure_metadata --urls --report_level warn --fail_level error --recursive "$DIR";
-        done < ../workflow_artifacts/changed_repositories.list
+        key: pip_cache_py_${{ matrix.python-version }}_gxy_${{ needs.setup.outputs.galaxy-commit }}
+    - uses: mvdbeek/planemo-discover-action@master
+      id: discover-repositories
```
Comment (bernt-matthias) [inline .github/workflows/pr.yaml:89]: I think here something like

```
with:
    changed-repositories: ${{ needs.setup.outputs.changed-repositories }}
    lint-tools: true
```

is missing?

File: .github/workflows/pr.yaml [hunk 2]
```diff
@@ -345,26 +250,19 @@ jobs:
...
-      run: planemo test_reports tool_test_output.json --test_output tool_test_output.html
-    - name: Copy artifacts into place
-      run: |
-        mkdir upload
-        mv tool_test_output.json tool_test_output.html upload/
+    - uses: mvdbeek/planemo-discover-action@master
+      with:
+         changed-repositories: ${{ needs.setup.outputs.changed-repositories }}
+         combine-outputs: true
+         html-report: true
```
Comment (bernt-matthias) [inline .github/workflows/pr.yaml:265]: Can we set the path to the generated html report explicitly in the action?

File: .github/workflows/pr.yaml [hunk 3]
```diff
@@ -50,51 +50,21 @@ jobs:
...
-      run: echo "COMMIT_RANGE=${{ github.event.before }}.." >> $GITHUB_ENV
-    - name: Set commit range (pull request)
-      if: github.event_name == 'pull_request'
-      run: echo "COMMIT_RANGE=HEAD~.." >> $GITHUB_ENV
-    - name: Planemo ci_find_repos
-      run: planemo ci_find_repos --changed_in_commit_range $COMMIT_RANGE --exclude packages --exclude deprecated --exclude_from .tt_skip --output changed_repositories.list
-    - name: Show repo list
-      run: cat changed_repositories.list
-    - uses: actions/upload-artifact@v2.0.1
+    - uses: mvdbeek/planemo-discover-action@master
```
Comment (bernt-matthias) [inline .github/workflows/pr.yaml:56]: Would like to have `name: ...` for those steps.

**Review by bernt-matthias — COMMENTED** [review]
In the output there are some distracting outputs: 

```
+ '[' false '!=' false ']'
+ '[' tools/add_input_name_as_column == '' ']'
+ echo tools/add_input_name_as_column
+ '[' false == true ']'
+ '[' false == true ']'
+ '[' false == true ']'
+ '[' '' == true ']'
```

Also I'm not sure if caching works yet. The output suggests that pip downloads the packages in each job.

**Comment by bernt-matthias** [comment]
I'm in favor of `galaxyproject/planemo-action`

**Comment by bernt-matthias** [comment]
Wondering if this can be readily adapted for the CI workflow. There a few wee differences, e.g. the grouping of tools.

File: .github/workflows/pr.yaml (hunk 2 above)
Comment (mvdbeek) [reply .github/workflows/pr.yaml:265]: Sure, it's just a bash script: https://github.com/mvdbeek/planemo-discover-action/blob/master/planemo_ci_actions.sh

**Comment by mvdbeek** [comment]
> Wondering if this can be readily adapted for the CI workflow. There a few wee differences, e.g. the grouping of tools.

I'm sure we can add this, I just based the script on what was on the master branch at the time.

**Comment by bernt-matthias** [comment]
closed https://github.com/galaxyproject/tools-iuc/pull/3437

**Pull request closed without merging**